sympy/calculus/util.py

```diff
@@ -1649,3 +1552,133 @@ def union(self, other):
 
 # setting an alias for AccumulationBounds
 AccumBounds = AccumulationBounds
+
+# Hooks
+def Add_hook(*args, **options):
+    evaluate = options.get('evalaute', global_parameters.evaluate)
+    if not evaluate:
+        return None
+
+    extras, accbnds, others = [], [], []
+    for a in args:
+        if isinstance(a, MatrixExpr):
+            extras.append(a)
+        elif isinstance(a, AccumBounds):
+            accbnds.append(a)
+        else:
+            others.append(a)
+    extra = Add(*extras, evaluate=True)
+    coeff = Add(*others, evaluate=True)
+
+    if coeff is S.NaN:
+        return S.NaN
+
+    # Combine the AccumBounds
+    for i,a in enumerate(accbnds):
+        if i == 0:
+            accbnd = a
+        else:
+            accbnd = AccumBounds(
+                Add(accbnd.min, a.min, evaluate=True),
+                Add(accbnd.max, a.max, evaluate=True))
+
+    # Combine coeff and accbnd
+    def _add_combine(self, other):
+        if other is S.Infinity and self.min is S.NegativeInfinity or \
+                other is S.NegativeInfinity and self.max is S.Infinity:
+            return AccumBounds(-oo, oo)
+        elif other.is_extended_real:
+            if self.min is S.NegativeInfinity and self.max is S.Infinity:
+                return AccumBounds(-oo, oo)
+            elif self.min is S.NegativeInfinity:
+                return AccumBounds(-oo, self.max + other)
+            elif self.max is S.Infinity:
+                return AccumBounds(self.min + other, oo)
+            else:
+                return AccumBounds(Add(self.min, other), Add(self.max, other))
+        return Add(self, other, ignore_hook=True, evaluate=False)
+    result = _add_combine(accbnd, coeff)
+
+    # Every combinables are combined.
+    return Add(result, extra, ignore_hook=True, evaluate=False)
+
+def Mul_hook(*args, **options):
+    evaluate = options.get('evalaute', global_parameters.evaluate)
```

Comment:
Misspelled evaluate.

A test for this case should be added if there isn't already one.

Reply:
Got it done. Thanks.